lib/puppet/transaction/report.rb

```diff
@@ -175,6 +181,11 @@ def finalize_report
   end
 
   # @api private
+  def get_settings
+    {"noop" => (Puppet[:noop] || false)}
```

Comment:
Shouldn't puppet defaults handle the defaulting already to false? Seems like we're going to be rewriting the default settings system inherit in puppet if we go down this road of providing defaults here also.